src/mono/mono/eglib/giconv.c

```diff
@@ -41,18 +38,6 @@
 #define UNROLL_DECODE_UTF8 0
 #define UNROLL_ENCODE_UTF8 0
 
-typedef int (* Decoder) (char *inbuf, size_t inleft, gunichar *outchar);
-typedef int (* Encoder) (gunichar c, char *outbuf, size_t outleft);
-
-struct _GIConv {
-	Decoder decode;
-	Encoder encode;
-	gunichar c;
-#ifdef HAVE_LIBICONV
-	iconv_t cd;
-#endif
-};
-
 static int decode_utf32be (char *inbuf, size_t inleft, gunichar *outchar);
 static int encode_utf32be (gunichar c, char *outbuf, size_t outleft);
```

Comment:
the decode_utf16 and decode_utf32 functions are unused too now, I didn't remove them here but let me know if you think they aren't really needed.